Update sharding.md example code

iAmZiNc opened the pull request:
Update sharding manager example code to include examples and related reference material for:
ShardManager options
Making use of the shardCreate event to log shard spawns

Commit:
Update sharding.md example code

Update sharding manager example code to include examples and related reference material for:
ShardManager options
Making use of the shardCreate event to log shard spawns

Commit:
Add some semicolons

I don't use em, but the guide does

Commit:
Explanation and examples

Add 3 new sections for data processing in sharded environment:
Sharing Information Between Shards
FetchClientValues
BroacastEval

Commit:
v12 code snuck in

Commit:
Update per York's feedback

Add:
Friendly descriptions and introductions to topics.
Other useful console.log() output examples.
Examples of things that each function *does not* support retreiving.

YorkAARGH commented:
The only thing I can suggest is make a note of what `this` is in regards to the code, other than that it looks good to me!.

Commit:
Update sharding.md

Add:
Some periods.
Re-state importance of broadcastEval() being run in the context of the client.

ghost commented:
> 
> 
> The only thing I can suggest is make a note of what `this` is in regards to the code, other than that it looks good to me!.

Adjusted

ghost commented:
- [ ] Add internal sharding example + explanation
- [ ] Add better explanation of returned values from broadcastEval()

Commit:
Update sharding.md

See 1b05c8600a4ca89b50c591486a1309aa2b2207f5
